1_wqp_pull/src/wqp_pull.R

```diff
@@ -111,45 +96,140 @@ filter_partitions <- function(partitions_ind, pull_task) {
 }
 
 # pull a batch of WQP observations, save locally, return .tind file
-get_wqp_data <- function(ind_file, partition, wqp_pull_params) {
-
+get_wqp_data <- function(data_file, partition, wqp_pull_params, verbose = TRUE) {
+  
   # prepare the arguments to pass to readWQPdata
-  wqp_args <- wqp_pull_params
-  wqp_args$characteristicName <- as.character(unlist(wqp_pull_params$characteristicName))
+  wqp_args <- list()
+  wqp_args$characteristicName <- wqp_pull_params$characteristicName$temperature
   wqp_args$siteid <- partition$MonitoringLocationIdentifier
-
   # do the data pull
-  wqp_dat_time <- system.time({
-    wqp_dat <- do.call(dataRetrieval::readWQPdata, wqp_args[c('siteid', 'characteristicName')])
-  })
-  message(sprintf(
-    'WQP pull for %s took %0.0f seconds and returned %d rows',
-    partition$PullTask[1],
-    wqp_dat_time['elapsed'],
-    nrow(wqp_dat)))
-
+  # first pull using readWQPdata, then if that fails, try POST
+  
+ 
+  getWQP <- function(wqp_args) {
+    wqp_dat_time <- tryCatch(
+      {
+        time_start <- Sys.time()
+        wqp_dat <- wqp_POST(wqp_args)
```

Comment:
Yes - sometimes you do. It's mostly because the call times out (according to Jim) - but is successful once the call is made again. 

It didn't look like any of the calls were made with readWQPdata.